### ISSUE TITLE
Add support for returning snippets with search results

### ISSUE THREAD

**Issue by coryschires** [kidpollo/tanker#7]
I'd like to fetch results which include both a reference to my model instance as well as a snippet showing where the search term appeared in the indexed doc.

Looking thru the index tank docs, there's built in support for snippets functionality (also saw some reference to it in the tanker code). After poking around, however, I can't seem to solve a few problems:

1) Tanker returns only model instances. That's good – and half of what I want. I would also like it to return the snippets (preferably as a virtual attribute so I can call @post.snippets along with the other attributes already being returned thru the model instance)

2) Even when sprinkling trace statements throughout the Tanker code, I can't seem to figure out where/when/why my searches fail to fetch snippets. The snippets functionality works fine when strictly following the basic ruby tutorial in the index tank docs. Something in the Tanker code seems to be swallowing up the snippets (or preventing them from being fetched in the first place?).

If I'm missing something basic, let me know. Otherwise, I think this would be a feature worth adding. I'd be willing to help out. I just need a little direction as I'm not sure at what point the snippets are getting lost. 

Thank for the work on this gem. It beats the pants off thinking_tank.

Cory.

**Comment by kidpollo** [comment]
Hey Cory!

When I started the gem snippets where still kinda beta so I couldn't get to it but now that we have direct contact with the Index Tank peeps we have first look into what is going!

I will work on this and other features I want to add for the end of this month. I will try to release every month a new version in the meantime I will let you know if I have it before.

Fell free to suggest your code for this solution in the meantime.

**Comment by coryschires** [comment]
Awesome. I'll let you know if I come up with anything.

Thanks!
Cory.

**Comment by bhellman1** [comment]
Hoping to check-in on snippets. Are snippet available? I'm trying with:

```
@posts = Post.search_tank(search_query, :snippet => 'text')
```

No error, but no snippet. Thanks

**Comment by kidpollo** [comment]
I am close to finishing this! Things at my say job have been crazy busy so I have not had to much time to work on this. Also I gave a talk on Tanker in Rails conf so I was preparing for that. I am really exited for the things to come. It should not be much longer but what you can be sure is that it will be awesome and fast!

**Comment by bhellman1** [comment]
@kidpollo nice! looking forward to it.Is your tanker talk available online?

**Comment by kidpollo** [comment]
FINALLY Sippeting and fetching attributes is done, sorry for the holdup. It has been a crazy past weeks. I also had been working on integration and benchmark specs but I just released version 1.1 of tanker so go ahead and se the updated readme on how to start using this new feature :D

**Comment by coryschires** [comment]
You're a great man, sir! I plan to give it a try during the next few days. Thanks a lot!
